### PR TITLE
Bump Go version to 1.22 in dev containers config

### PR DESCRIPTION
**Pull request by sebhtml**
# Context

In the [README](https://github.com/prebid/prebid-server/blob/v2.31.0/README.md?plain=1#L47), it is written that:
> Prebid Server requires Go version 1.21 or newer.

However, in the [Dev Containers configuration](https://github.com/prebid/prebid-server/blob/v2.31.0/.devcontainer/devcontainer.json#L9), Go 1.20 is configured:
> 			"VARIANT": "1.20",

# Consequence

A natural consequence of that is the following compile error is obtained when one opens the project with Dev Containers with Visual Studio Code (tested with VS Code 1.93.1):

> could not import maps (no required module provides package "maps") compiler BrokenImport

# Root Cause Analysis

The package "maps" does not exist in Go 1.20. See https://pkg.go.dev/maps?tab=versions
If there is a table with nothing on it and you ask someone to eat the apple that is on the table, there will be consequences.
(Feedback from the environment)

# Solution

A no-brainer solution is to make the Dev Containers configuration congruent with the README file.

### CODE REVIEW

**Comment by SyntaxNode** [comment]
Nice catch. @bsardo missed this when upgrading to Go 1.22. While we support 1.21, the dockerfile is building with 1.22.3. Do you think we should update the dev container to the same version?

**Comment by bsardo** [comment]
> Nice catch. @bsardo missed this when upgrading to Go 1.22. While we support 1.21, the dockerfile is building with 1.22.3. Do you think we should update the dev container to the same version?

Oops. Yes let's update this to 1.22.

**Comment by sebhtml** [comment]
> > Nice catch. @bsardo missed this when upgrading to Go 1.22. While we support 1.21, the dockerfile is building with 1.22.3. Do you think we should update the dev container to the same version?
> 
> Oops. Yes let's update this to 1.22.

Hello @bsardo 

Should I update to 1.22 in Dev Containers and in the README ?

Thank you.

**Comment by bsardo** [comment]
> Hello @bsardo
> Should I update to 1.22 in Dev Containers and in the README ?

Hi @sebhtml, yes please. That would be great.

**Comment by sebhtml** [comment]
> > Hello @bsardo
> > Should I update to 1.22 in Dev Containers and in the README ?
> 
> Hi @sebhtml, yes please. That would be great.

Hello @bsardo,

It's done. See commit https://github.com/prebid/prebid-server/pull/3983/commits/5aae4cd81db123b46335f4bbd85475352a030680 .


Nice catch. @bsardo missed this when upgrading to Go 1.22. While we support 1.21, the dockerfile is building with 1.22.3. Do you think we should update the dev container to the same version?

Oops. Yes let's update this to 1.22.

Hello @bsardo

Should I update to 1.22 in Dev Containers and in the README ?

Thank you.

File: README.md [hunk 1]
```diff
@@ -44,7 +44,7 @@ Ensure that you deploy the `/static` directory, as Prebid Server requires those
 
 ## Developing
 
-Prebid Server requires [Go](https://go.dev) version 1.21 or newer. You can develop on any operating system that Go supports; however, please note that our helper scripts are written in bash.
+Prebid Server requires [Go](https://go.dev) version 1.22 or newer. You can develop on any operating system that Go supports; however, please note that our helper scripts are written in bash.
```
Comment (bsardo) [inline README.md:47]: Gah my mistake. It appears the README does not need to be updated. 1.21 is actually correct here as you can see that is what we declare in `go.mod`. We are currently supporting 1.21 and 1.22. The dev container and our github actions though should all run with 1.22. Can you undo this change?

Comment (sebhtml) [reply README.md]: Hi @bsardo 

Sure.

See "Update README.md to put back Go 1.21" -> https://github.com/prebid/prebid-server/pull/3983/commits/caf0c5694501c7c3799db4eff33f6da5ca5212d6

**Review by bsardo — APPROVED** [review]
